pandas/core/arrays/interval.py

```diff
@@ -777,6 +777,17 @@ def take(self, indices, allow_fill=False, fill_value=None, axis=None,
 
         return self._shallow_copy(left_take, right_take)
 
+    def where(self, cond, other):
```

Comment:
Would be nice to have `IntervalIndex` use this implementation instead of the naive object array based implementation that it currently uses.  Can certainly leave that for a follow-up PR though, and I'd be happy to do it.